backend/MoleculeOSSite/Controllers/AccountController.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using MoleculeOSSite.Entities;
+using MoleculeOSSite.ModelsDTO;
+using MoleculeOSSite.Services;
+
+namespace MoleculeOSSite.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class AccountController : ControllerBase
+    {
+        private readonly IAccountService _accountService;
+
+        public AccountController(IAccountService accountService)
+        {
+            _accountService = accountService;
+        }
+
+
+        [HttpPost("register")]
+        public ActionResult RegisterUser([FromBody]User user)
+        {
+            _accountService.RegisterUser(user);
```

Comment:
Nie używaj User tutaj, bo to jest entity, a nie dane potrzebne do rejestrowania. Nie mieszaj logiki rejestrowania oraz ogólnej do opisywania modeli w bazie. Przez to kod jest bardzo nie czytelny (np: PasswordHash to jednocześnie hasło i hash hasła).
Zamiast tego stwórz dodatkowe dto np: UserDTO, RegisterDataDTO czy po prostu RegisterDTO (chyba najlepsza jest ta trzecia nazwa) oraz zmapuj (przypisz) dane z tego nowego dto do Usera w serwisie.